lib/srv/monitor_test.go

```diff
@@ -205,11 +205,9 @@ func TestMonitorStaleLocks(t *testing.T) {
 	case <-time.After(15 * time.Second):
 		t.Fatal("Timeout waiting for LockWatcher loop check.")
 	}
-	select {
-	case asrv.LockWatcher.StaleC <- struct{}{}:
-	default:
-		t.Fatal("No staleness event should be scheduled yet. This is a bug in the test.")
-	}
+	// StaleC is listened by multiple goroutines, so we need to close to ensure
+	// that all of them are unblocked.
+	close(asrv.LockWatcher.StaleC)
```

Comment:
If all we do is close `StaleC` rather than pushing into it then it should be made an unbuffered channel in `lib/services.newResourceWatcher`, just so we don't keep wondering why it's a buffered channel in the future.